### ISSUE TITLE
Clarification on Number of Training Steps

### ISSUE THREAD

**Issue by alberthli** [jishengpeng/WavTokenizer#65]
In the paper and in the config comment, you state that you train the model for 1M steps each for the discriminator and generator. However, the config itself uses
```
max_steps: 20000000  # 20M, not 2M!
```
Could you clarify which of these numbers is a typo?

**Comment by jishengpeng** [comment]
> In the paper and in the config comment, you state that you train the model for 1M steps each for the discriminator and generator. However, the config itself uses
> 
> ```
> max_steps: 20000000  # 20M, not 2M!
> ```
> 
> Could you clarify which of these numbers is a typo?

Thank you for your attention. We set an upper limit of 2 million steps for training. And in practice, the training process is often terminated earlier based on observations from TensorBoard.

**Comment by alberthli** [comment]
So you mean to say that the number `20000000` (20 million, not 2 million) in the config is always set, but you terminate early, usually around 2 million steps instead? And do you just manually terminate the run rather than using some automated condition?

**Comment by jishengpeng** [comment]
> So you mean to say that the number `20000000` (20 million, not 2 million) in the config is always set, but you terminate early, usually around 2 million steps instead? And do you just manually terminate the run rather than using some automated condition?

2 million, not 20 million.

**Comment by alberthli** [comment]
I'm not sure I understand. [This number](https://github.com/jishengpeng/WavTokenizer/blob/afdec2512c0778746250f6fc40d4bac7ff82b742/configs/wavtokenizer_smalldata_frame40_3s_nq1_code4096_dim512_kmeans200_attn.yaml#L87) in your config is not 2 million, it is 20 million (there are 7, not 6 zeros). Are you saying that the number specified in the paper and the comments is wrong, or that the config is wrong? These two numbers are not consistent with each other.

**Comment by jishengpeng** [comment]
> I'm not sure I understand. [This number](https://github.com/jishengpeng/WavTokenizer/blob/afdec2512c0778746250f6fc40d4bac7ff82b742/configs/wavtokenizer_smalldata_frame40_3s_nq1_code4096_dim512_kmeans200_attn.yaml#L87) in your config is not 2 million, it is 20 million (there are 7, not 6 zeros). Are you saying that the number specified in the paper and the comments is wrong, or that the config is wrong? These two numbers are not consistent with each other.

We have further updated the config for the better understanding. Thank you.

**Comment by alberthli** [comment]
Thank you for updating the config - did you use 2M or 20M when training the models presented in the paper? This affects things like the learning rate scheduler.
